test/queryRunner.test.ts

```diff
@@ -727,6 +727,153 @@ suite('Query Runner tests', () => {
+                ]
+            }
+        };
+        process.env['LANG'] = 'C';
+
+        let testRange: ISlickRange[] = [{fromCell: 0, fromRow: 0, toCell: 1, toRow: 2},
+                                        {fromCell: 1, fromRow: 1, toCell: 2, toRow: 4}];
+
+        let result: QueryExecuteCompleteNotificationResult = {
+            ownerUri: testuri,
+            batchSummaries: [{
+                hasError: false,
+                id: 0,
+                selection: {startLine: 0, endLine: 0, startColumn: 3, endColumn: 3},
+                resultSetSummaries: <ResultSetSummary[]> [{
+                    id: 0,
+                    rowCount: 5,
+                    columnInfo: [
+                        { columnName: 'Col1' },
+                        { columnName: 'Col2' },
+                        { columnName: 'Col3' }
+                    ]
+                }],
+                executionElapsed: undefined,
+                executionStart: new Date().toISOString(),
+                executionEnd: new Date().toISOString()
+            }]
+        };
+
+        setup(() => {
+            testSqlToolsServerClient.setup(x => x.sendRequest(TypeMoq.It.isAny(),
+                                                            TypeMoq.It.isAny())).callback(() => {
+                                                                // testing
+                                                            }).returns(() => { return Promise.resolve(testresult); });
+            testStatusView.setup(x => x.executingQuery(TypeMoq.It.isAnyString()));
+            testStatusView.setup(x => x.executedQuery(TypeMoq.It.isAnyString()));
+            testVscodeWrapper.setup( x => x.logToOutputChannel(TypeMoq.It.isAnyString()));
+            testVscodeWrapper.setup(x => x.clipboardWriteText(TypeMoq.It.isAnyString())).callback(() => {
+                                                                // testing
+                                                            }).returns(() => { return Promise.resolve(); });
+
+        });
+
+        // ------ Copy tests with multiple selections  -------
+        test('Correctly copy pastes a selection', (done) => {
+            let configResult: {[key: string]: any} = {};
+            configResult[Constants.copyIncludeHeaders] = false;
+            setupWorkspaceConfig(configResult);
+
+            let queryRunner = new QueryRunner(
+                testuri,
+                testuri,
+                testStatusView.object,
+                testSqlToolsServerClient.object,
+                testQueryNotificationHandler.object,
+                testVscodeWrapper.object
+            );
+            queryRunner.uri = testuri;
+            queryRunner.copyResults(testRange, 0, 0).then(() => {
+                testVscodeWrapper.verify<void>(x => x.clipboardWriteText(TypeMoq.It.isAnyString()), TypeMoq.Times.once());
```

Comment:
This test would be improved by actually checking the pasted value matches what you expect